### DISCUSSION TITLE
Anyone try Raspbian 64-bit?

### DISCUSSION THREAD

**Discussion by MadTooler** [winder/Universal-G-Code-Sender#2092] [Help]
Has anyone tried running UGS on a raspberry pi 4b with the 64-bit official OS?

**Comment by MadTooler** [comment]
Welllll, I tried and cannot get it to run.  I installed the latest raspberry 64 OS and do not find the "G2 GL (Fake KMS)" driver option noted in the UGS wiki install any longer.  I do not know if that is related to the issue or not, but I have yet to get UGS to load.  It does not crash the pi, but also never does anything after I tell the current 2.0.12 or the pre-release 2.0.13 ugsplatform script to run.   I also tried having it execute in terminal.  It flashed the terminal screen and closes it too fast to see anything.

Is there some deeper graphics tweaking needed or is this a no go for 64 bit raspberry OS?

**Comment by MadTooler** [comment]
Does "apt-get install openjdk-8-jdk" need to be installed for ugs 2.x? 

I initially only ran through the steps noted in the wiki for the older version of ugs 2.0.7 and did not install the openjdk explicitly as noted there for the classic version.  Just in case that was related, I have tried to run the line "apt-get install openjdk-8-jdk" and get an error that it could not find it.  Is this related, or again, does ugs not yet run on rpi 64 OS?

**Comment by breiler** [comment]
For the Platform edition of UGS you generally don't need install Java as it comes with the software. I don't have an Raspberry pi 4 so I can't test this myself and have difficulties to help you.

Java is not included in the Classic version so you need to install it separately. The Java 8 that we refer to in the documentation doesn't seem to be available anymore. Try and install `openjdk-11-jdk` instead.

**Reply by MadTooler** [reply]
Thanks for the reply.  I hope you have had some peaceful holidays.

I am only trying to use the Platform edition.  I tried the version 11 java just in case, since I noted the 64 bit OS was significantly smaller install than the 32 bit OS, but it did not make any difference.  That was to be expected since the platform edition came with. 

From a terminal, within the ugs/ugsplatform-pi/bin folder, I ran "bash ugsplatform".  It gave me an error
`./../platform/lib/nbexec: line 430:  /home/(myPi'sNameHere)/ugs/ugsplatform-pi/jdk/jdk8u275-b01-jre/bin/java: No such file or directory`

I checked and the "java" file exists in that location.  Is this any better indicator of what may be wrong?

Should I open an issue for this?

**Reply by breiler** [reply]
Is your account name really `(myPi'sNameHere)`?

While special characters such as `'` and `(` in directory names shouldn't be a problem for modern softwares, Netbeans is sensitive to it. A good practice is to have account names with characters `a-z`.

**Reply by MadTooler** [reply]
> Is your account name really `(myPi'sNameHere)`?

No.  I was emphasizing that my username is not pi and not sharing my actual username for project related reasons.  The new OS installations ask for new usernames now as a standard for security.  I wouldn't think not "pi" is an issue for any of the code of UGS, but is it?

Did the error I noted point to anything helpful?
